### DISCUSSION TITLE
Speech2Text - Help needed with alphanumeric words

### DISCUSSION THREAD

**Discussion by rohan25k** [deepgram/community#745] [General help] [answered]
Hi
I am trying out your nova2-medical model. I wanted to know if there is a way to get all alphabet response from this model?
This model is transcribing as follows -> "Vtamin B12", "Code: C34D2", "Patient Id: 4C4", "D589"
What I want is something like -> "Vitamin B Twelve", "Code: C Three Four Zero Two", "Patient Id: Four C Four", "D Five Hundred amd Eighty Nine" etc In this, if "12" will be transcried as "Twelve" or "One Two" will be decided by speech or how it has been said.

I want no digits/numbers in my transcript.

Thank You

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @rkchamp25, I recommend trying with `smart_format=false`, which will generally avoid this formatting of numbers and other entities, and just give you the written-out transcript.

**Reply by rohan25k** [reply]
@jkroll-deepgram 
Sorry, I forgot to mention that I have tried both ways with smart_format = True and False, this only changes the numbers which are separate for example when smart_format = True -> "four ive nine" -> "459" and if it is false it stays as "four five nine" which is good. But for the case of alphanumeric words, smart format doesn't make a difference, for example smart_format=True or False -> "a12fg" -> "a12fg" but what I want is "a one two f g".

Model - Nova-2-Medical

**Reply by jkroll-deepgram** [reply, marked as answer]
@rkchamp25, thanks for the clarification. Given that info, I recommend post-processing, as I don't believe Deepgram's formatting allows for that case. For instance, I've used the [Python num2words package](https://pypi.org/project/num2words/), and there are similar packages in other programming languages.

**Reply by rohan25k** [reply]
With Nova-2-Medical alphanumeric, stay alphanumeric.
With Nova-2, alphanumeric -> only as alphabets just like I want. 
Please see why this is the case @jkroll-deepgram @jpvajda 
I want to use medical model but for now I am transcribing with nova-2 for all the audios that have some alphanumeric words.

**Reply by jkroll-deepgram** [reply]
If nova-2 is giving you the formatting that you need, and the medical model isn't, then you're able to either continue using nova-2 or post-processing the nova-2 medical output. 

The medical model formatting is baked into the model and purpose-built for the most commonly preferred entity formatting that we've heard from our medical customers. That being said, I'll pass on your feedback to the team.
